src/hermit/mod.rs

```diff
@@ -28,6 +28,547 @@ pub type wctype_t = i64;
 pub type regoff_t = size_t;
 pub type off_t = c_long;
 
+pub type sa_family_t = u8;
+pub type socklen_t = u32;
+pub type in_addr_t = u32;
+pub type in_port_t = u16;
+pub type time_t = i64;
+pub type suseconds_t = i64;
+pub type nfds_t = usize;
+
+s! {
+/// `timespec` is used by `clock_gettime` to retrieve the
+/// current time
+pub struct timespec {
+    /// seconds
+    pub tv_sec: i64,
+    /// nanoseconds
+    pub tv_nsec: i64,
+}
+
+pub struct in_addr {
+    pub s_addr: u32,
+}
+
+pub struct in6_addr {
```

Comment:
We might want to raise the alignment here:

```suggestion
#[repr(align(4))]
pub struct in6_addr {
```